zebra-chain/src/amount.rs

```diff
@@ -227,6 +227,12 @@ impl<C1, C2> PartialEq<Amount<C2>> for Amount<C1> {
     }
 }
 
+impl<C> PartialEq<i64> for Amount<C> {
+    fn eq(&self, other: &i64) -> bool {
+        self.0.eq(other)
+    }
+}
```

Comment:
Let's also `impl<C> PartialEq<Amount<C>> for i64`?

Otherwise it's really confusing when you compare around the other way.

Reply:
Done, but I wasn't sure if it was best to do `other.eq(self)` or actually implementing it as `self.eq(&other.0)`. I ended up going with the latter, but I can change it to the former if that's better.

Reply:
In this case, it doesn't really matter.

But in general we try to re-use complex eq implementations.